src/plugins/ui_feature/autofill.ts

```diff
@@ -238,6 +246,108 @@ export class AutofillPlugin extends UIPlugin {
     }
   }
 
+  private collectBordersData(
+    data: Omit<AutoFillCellCommand, "type">,
+    borders: Record<UID, Border>,
+    bordersPositions: Record<number, Zone[]>
+  ) {
+    const id = getItemId(data.border, borders);
+    if (!(id in bordersPositions)) {
+      bordersPositions[id] = [];
+    }
+    bordersPositions[id].push(positionToZone({ col: data.col, row: data.row }));
+  }
+
+  private collectConditionalFormatsData(
+    data: Omit<AutoFillCellCommand, "type">,
+    cfNewRanges: Record<UID, string[]>
+  ) {
+    const { originCol, originRow, col, row } = data;
+    const sheetId = this.getters.getActiveSheetId();
+    const cfOrigin = this.getters.getRulesByCell(sheetId, originCol, originRow);
+    const xc = toXC(col, row);
+    for (const cf of cfOrigin) {
+      if (!(cf.id in cfNewRanges)) {
+        cfNewRanges[cf.id] = [];
+      }
+      cfNewRanges[cf.id].push(xc);
+    }
+  }
+
+  private collectDataValidationsData(
+    data: Omit<AutoFillCellCommand, "type">,
+    dvNewZones: Record<UID, Zone[]>
+  ) {
+    const { originCol, originRow, col, row } = data;
+    const sheetId = this.getters.getActiveSheetId();
+    const cellPosition = { sheetId, col: originCol, row: originRow };
+    const dvOrigin = this.getters.getValidationRuleForCell(cellPosition);
+    if (!dvOrigin) {
+      return;
+    }
+    if (!(dvOrigin.id in dvNewZones)) {
+      dvNewZones[dvOrigin.id] = [];
+    }
+    dvNewZones[dvOrigin.id].push(positionToZone({ col, row }));
+  }
+
+  private autofillCell(data: Omit<AutoFillCellCommand, "type">) {
+    this.dispatch("UPDATE_CELL", {
+      sheetId: this.getters.getActiveSheetId(),
+      col: data.col,
+      row: data.row,
+      content: data.content || "",
+      style: data.style || null,
+      format: data.format || "",
+    });
+    this.dispatch("AUTOFILL_CELL", data);
+  }
+
+  private autofillBorders(borders: Record<UID, Border>, bordersPositions: Record<number, Zone[]>) {
+    for (const [borderId, border] of Object.entries(borders)) {
+      this.dispatch("SET_BORDER_TARGET", {
+        sheetId: this.getters.getActiveSheetId(),
+        border,
+        target: futureRecomputeZones(bordersPositions[Number(borderId)]),
+      });
+    }
+  }
+
+  private autofillConditionalFormats(cfNewRanges: Record<UID, string[]>) {
+    for (const [cfId, changes] of Object.entries(cfNewRanges)) {
+      const sheetId = this.getters.getActiveSheetId();
+      const cf = this.getters.getConditionalFormats(sheetId).find((cf) => cf.id === cfId);
+      if (!cf) {
+        continue;
+      }
+      const newCfRanges = this.getters.getAdaptedCfRanges(sheetId, cf, changes, []);
+      if (newCfRanges) {
+        this.dispatch("ADD_CONDITIONAL_FORMAT", {
+          cf: deepCopy(cf),
```

Comment:
the property "cf.ranges" (string)  is not correct, it looks like it contains the previous version of the ranges (zones)

Reply:
`ranges` is ignored, the `cf` property is typed ad `cf: Omit<ConditionalFormat, "ranges">;`
So I simply removed it ;-)